Guide: monitoring performance after ZEIT deploy with Calibre (#1649)
zeit/docs: pages/guides/monitoring-performance-with-calibre-and-zeit-now.mdx

import Guide from '~/components/layout/guide'
import Link from '~/components/text/link'
import Note from '~/components/text/note'
import { Image } from '~/components/media'
import Caption from '~/components/text/caption'
import { IconCard } from '~/components/card'
import { List } from '~/components/list'
import GitHubIcon from '~/components/icons/github'

export const meta = {

>   title: 'Monitor performance with Calibre and ZEIT Now',

Reviewer: ```suggestion
  title: 'Monitor Performance with Calibre and ZEIT Now',
```

  description:
    'Receive performance reports directly in your GitHub Pull Requests, following a ZEIT Now deploy.',
  published: '2020-02-12T03:50:35Z',
  authors: ['thefoxis'],
  url: '/guides/monitoring-performance-with-calibre-and-zeit-now',
  editUrl: 'pages/guides/monitoring-performance-with-calibre-and-zeit-now.mdx',
  lastEdited: '2020-02-17T00:30:35.000Z'
}

> [Calibre](https://calibreapp.com/) is a comprehensive monitoring platform that makes it easy to uncover and address performance issues. In this guide, you will learn how receive performance [Pull Request Reviews](https://calibreapp.com/pull-request-reviews) directly in your pull requests following [ZEIT Deployment Previews](https://zeit.co/features/deployment-previews).

Reviewer: ```suggestion
[Calibre](https://calibreapp.com/) is a comprehensive monitoring platform that makes it easy to uncover and address performance issues. In this guide, you will learn how to receive performance [Pull Request Reviews](https://calibreapp.com/pull-request-reviews) directly in your pull requests following [ZEIT Deployment Previews](https://zeit.co/features/deployment-previews).
```

> ## Step 1: Deploy your site or app with ZEIT Now

Reviewer: ```suggestion
## Step 1: Deploy with ZEIT Now
```

> To deploy your project with a [ZEIT Now for Git Integration](/docs/v2/git-integrations), make sure your created project has been pushed to a Git repository.

Reviewer: ```suggestion
To deploy your project with the [ZEIT Now for GitHub Integration](/docs/v2/git-integrations/zeit-now-for-github), make sure your created project has been pushed to a Git repository.
```

[Import the project](/new) into ZEIT Now from GitHub:

<List columnsDesktop={3} columnsTablet={3} columnsMobile={1}>
  <IconCard label="GitHub" icon={<GitHubIcon />} href="/new?tab=github" plus />
</List>

After your project has been imported, all subsequent pushes to branches will generate [Preview Deployments](/docs/v2/platform/deployments#preview), and all changes made to the default branch (commonly `master`) will result in a [Production Deployment](/docs/v2/platform/deployments#production).

<Note type="warning">

>   While ZEIT supports GitHub, Bitbucket and GitLab, Calibre Pull Request Reviews

Reviewer: ```suggestion
  While ZEIT Now supports GitHub, Bitbucket, and GitLab, Calibre Pull Request Reviews
```

>   currently work with ZEIT and GitHub only. GitLab and Bitbucket are not

Reviewer: ```suggestion
  currently work with GitHub only. GitLab and Bitbucket are not
```

  supported.
</Note>

> ## Step 2: Add your site to Calibre

Reviewer: ```suggestion
## Step 2: Add Your Site to Calibre
```

To receive performance metrics and reports, start by [creating a Calibre account](https://calibreapp.com/sign-up) and adding sites to monitor. You will be prompted to add a name, provide a URL, choose from 14 global test locations, and set a testing schedule.

Calibre allows you to **check unlimited sites and pages**. Each site can be connected to a repository to receive Pull Request Reviews.

<Image
  src={`${
    process.env.ASSETS
  }/guides/monitoring-performance-with-calibre-and-zeit-now/calibre-add-site.png`}
  width={1536 / 2}
  height={892 / 2}
  shadow
/>
<Caption>Creating a new Site in Calibre.</Caption>

> If you’d like to learn more about how to successfully set up a performance monitoring system, read our [Getting Started](https://calibreapp.com/docs/get-started/guide) guide.

Reviewer: ```suggestion
If you’d like to learn more about how to successfully set up a performance monitoring system, read the Calibre [Getting Started](https://calibreapp.com/docs/get-started/guide) guide.
```

## Step 3: Connect Calibre to GitHub

> To enable Pull Request Reviews, go to **Site → Settings → Integrations** tab and click **Install** next to the GitHub logo. You will be redirected to GitHub to enable the integration for your organisation.

Reviewer: ```suggestion
To enable Pull Request Reviews, from the Calibre Dashboard, go to **Site → Settings → Integrations** tab and click **Install** next to the GitHub logo. You will be redirected to GitHub to enable the integration for your organization.
```

<Image
  src={`${
    process.env.ASSETS
  }/guides/monitoring-performance-with-calibre-and-zeit-now/calibre-github-integration.png`}
  width={1536 / 2}
  height={198 / 2}
  shadow
/>

> 

Reviewer: ```suggestion

<Caption>
  Installing the Calibre GitHub integration.
</Caption>

```

<Note type="success" label="hint">

>   Calibre requires access to commit statuses, deployments, checks and pull

Reviewer: ```suggestion
  Calibre requires access to commit statuses, deployments, checks, and pull
```

  requests. We strongly recommend only granting access to selected repositories
  you’re interested in tracking.
</Note>

> ### Select a comparison branch

Reviewer: ```suggestion
### Select a Comparison Branch
```

> After confirming the link between GitHub and Calibre, you will have to select a primary branch that will be the base for performance comparisons. This branch should be either deploying to production or a staging environment. In most cases, this might be `master` branch.

Reviewer: ```suggestion
After confirming the link between GitHub and Calibre, you will have to select a primary branch that will be the base for performance comparisons. This branch should be either deploying to production or a staging environment. In most cases, this might be the `master` branch.
```
